transaction/std/src/transaction_builder.rs

```diff
@@ -117,7 +115,26 @@ impl TransactionBuilder {
             })
             .collect();
 
-        let tx_prefix = TxPrefix::new(inputs, self.outputs.clone(), self.fee, self.tombstone_block);
+        // Sort outputs by public key.
+        self.outputs_and_shared_secrets
+            .sort_by(|(a, _), (b, _)| a.public_key.cmp(&b.public_key));
```

Comment:
This is the main change.

Reply:
we could sort by any field to solve the ordering problem... is there a performance reason to prefer `public_key`?